### PR TITLE
Hover Event Sending

### PR DESCRIPTION
**Pull request by jslavitz**
Sends hover events through to the framework.

### CODE REVIEW

File: shell/platform/android/io/flutter/view/FlutterView.java [hunk 1]
```diff
@@ -406,6 +407,13 @@ private int getPointerChangeForAction(int maskedAction) {
         if (maskedAction == MotionEvent.ACTION_CANCEL) {
             return kPointerChangeCancel;
         }
+        if (maskedAction == MotionEvent.ACTION_HOVER_MOVE ||
```
Comment (chinmaygarde) [inline shell/platform/android/io/flutter/view/FlutterView.java]: Why not add another phase explicitly for the "hover enter" like you did for the "exit"?

File: shell/platform/android/io/flutter/view/FlutterView.java [hunk 2]
```diff
@@ -372,10 +372,11 @@ public InputConnection onCreateInputConnection(EditorInfo outAttrs) {
     private static final int kPointerChangeCancel = 0;
     private static final int kPointerChangeAdd = 1;
     private static final int kPointerChangeRemove = 2;
-    private static final int kPointerChangeHover = 3;
+    private static final int kPointerChangeHoverMove = 3;
     private static final int kPointerChangeDown = 4;
     private static final int kPointerChangeMove = 5;
     private static final int kPointerChangeUp = 6;
+    private static final int kPointerChangeHoverExit = 8;
```
Comment (chinmaygarde) [inline shell/platform/android/io/flutter/view/FlutterView.java]: Doesn't this change need to be reflected in pointer.dart as well?

File: shell/platform/android/io/flutter/view/FlutterView.java [hunk 3]
```diff
@@ -533,8 +541,7 @@ public boolean onHoverEvent(MotionEvent event) {
 
         boolean handled = handleAccessibilityHoverEvent(event);
         if (!handled) {
-            // TODO(ianh): Expose hover events to the platform,
-            // implementing ADD, REMOVE, etc.
+            onTouchEvent(event);
```
Comment (Hixie) [inline shell/platform/android/io/flutter/view/FlutterView.java]: this doesn't seem to implement the intended protocol that we have designed for pointer events. The intended protocol is that we fire an "ADD" when we learn of the pointer, "MOVE" events while the pointer is moving, "DOWN" when it goes down/clicks, "UP" when it goes up, "MOVE" when it moves, and "REMOVE" when the pointer finally leaves the scene.

Comment (Hixie) [reply shell/platform/android/io/flutter/view/FlutterView.java]: see for instance https://master-docs-flutter-io.firebaseapp.com/flutter/gestures/PointerAddedEvent-class.html

Comment (jslavitz) [reply shell/platform/android/io/flutter/view/FlutterView.java]: So this desired functionality is going to be taken care of in the framework, not in the engine. Like we agreed, we'll simply add the mouse with a special pointer index of 0 and not ever remove it.

**Comment by jslavitz** [comment]
This is still a work in progress, we are waiting on clarification of specs from the google ux teams.

**Comment by jslavitz** [comment]
Sure!

**Pull request closed without merging**